core/generator.js

```diff
@@ -104,7 +105,10 @@ Blockly.Generator.prototype.workspaceToCode = function(workspace) {
       // Top-level blocks don't care about operator order.
       line = line[0];
     }
-    if (line) {
+
+    if (line == undefined) {
```

Comment:
The assumption here is that no generator would ever output an empty line. I don't think that is correct. A block might be disabled or otherwise intentionally not generating code. In some apps that have JavaScript-to-blocks functionality, there is a block that intentionally outputs an empty line to preserve the spacing in the code.

What you really want to test `this[block.type]` (or other language) undefined.
In fact, this is already tested inside `Generator.blockToCode()`, which I believe throws an error with the block name in the string (via `goog.assert`). What is doesn't do is traverse the block tree for all missing generator definitions.

Perhaps the best solution is to revert changes to the `generator.js` and add a preflight verification step inside each language `if` block of `Code.renderContent`. This should a function call that takes in the workspace and the language generator object, and in that function it should  tests each block.type in `workspace.getAllBlocks()` against the object keys of the generator. Then, if there is a missing block type, call `Blockly.alert(..)`.

Reply:
Okey, then, I will be working on it.

Reply:
How can I get  'language generator object'?

Reply:
Sorry for being late. My college semester has just started, I was preparing for it. What do you mean by 'language generator object' and 'object keys of the generator'? I could not understand. Thank you.